packages/dds/tree/src/shared-tree-core/sharedTreeCore.ts

```diff
@@ -311,7 +311,7 @@ export class SharedTreeCore<TEditor extends ChangeFamilyEditor, TChange> extends
 		// Empty context object is passed in, as our decode function is schema-agnostic.
 		const {
 			commit: { revision },
-		} = this.messageCodec.decode(content, {});
+		} = this.messageCodec.decode(this.serializer.decode(content), {});
```

Comment:
@tyler-cai-microsoft @CraigMacomber does this seem like a better place to decode the content on resubmit than shared object?

Reply:
i'd prefer this in shared object, as that is where all the handle encoding/decoding is done for ops. 

+ @markfields  @Abe27342

Reply:
Yeah. I think it's fair to be scared about that change, and I think for the purposes of this PR, it's probably OK to make the change here. However, to be consistent with the model where DDS extenders don't think about IFluidHandle encoding (other than being mindful that they exist and any serialization should use `SharedObject.serializer` rather than JSON.stringify), SharedObject / SharedObjectCore shouldn't be handing back the DDS an op that's different from what the DDS submitted.

Reply:
just to note, this is also a real shared tree bug when dealing with handles.

Reply:
If it's a real bug, would it make sense then to leave the decoding here for now and file a bug against shared tree to fix the handle handling later?

Reply:
ehh IMO an accurate categorization of things is that it's a bug with SharedObject's handle serialization scheme, but that bug happens to not affect most DDSes. SharedTree is an exception since it does some validation on ops that it processes / is asked to resubmit. So the right place to fix things is in SharedObject, but it would be reasonable to work around the more correct fix by making a more scoped change first and expanding it later. The "expanding it later" bit isn't really a bug to be filed against SharedTree though in this framing.

Staging things this way also gives us less risk that we'll have to back out the changes to add handles to fuzz tests, which is nice.